packages/dds/tree/src/simple-tree/schemaFactoryRecursive.ts

```diff
@@ -151,3 +152,34 @@ export type ValidateRecursiveSchema<
 		}[T["kind"]]
 	>,
 > = true;
+
+/**
+ * Workaround for fixing errors an issue with recursive ArrayNode schema exports.
+ * @remarks
+ * Importing a recursive ArrayNode schema via a d.ts file can produce an error like
```

Comment:
Is there an issue filed against TypeScript for this? Can we link to it from these docs?

Reply:
Random cases of recursive types not working is working as intended.

The fact that the d.ts generation breaks it could be an issue. Maybe we can suggest they stop the splitting out of base classes, which would fix this and the related API extractor issue? I'm not aware of a current issue for this, but finding or filing one and linking it here is a good idea.

Reply:
Issue filed and linked: https://github.com/microsoft/TypeScript/issues/59550